cmd/lifecycle/builder.go

```diff
@@ -48,26 +60,160 @@ func (b *buildCmd) Args(nargs int, args []string) error {
 }
 
 func (b *buildCmd) Privileges() error {
-	// builder should never be run with privileges
-	if priv.IsPrivileged() {
+	_, privGroup, _, err := b.readData()
+	if err != nil {
+		return err
+	}
+
+	hasPrivGroups := len(privGroup.Group) > 0
+
+	// builder should never be run with privileges if there aren't any stack buildpacks
+	if !hasPrivGroups && priv.IsPrivileged() {
 		return cmd.FailErr(errors.New("refusing to run as root"), "build")
```

Comment:
I am not sure this should actually be a failure. Maybe we can just drop privileges in this case. As a user I would be annoyed one build command succeeds and another (identical except for a single flag) created this error.